<?xml version="1.0"?>
<launch>
    <!-- 2D LIDAR CONNECTIONS -->
    <include file="$(find rplidar_ros)/launch/rplidar_c1.launch">
    </include>
        <node pkg="lidar" type="wheel_filter" name="wheel_filter">
            <remap from="filtered" to="scan"/>
        </node>

    <!-- DEPTH LIDAR CONNECTIONS -->
    <!-- <include file="$(find openni2_launch)/launch/openni2.launch">
    </include> -->

    <!-- ROS SERIAL CONNECTIONS -->
    <node pkg="rosserial_python" type="serial_node.py" name="serial_node1">
        <param name="port" value="/dev/ttyACM0"/>
        <param name="baud" value="115200"/>
    </node>

    <!-- IF YOU NEED A CONNECTING 2 DEVICE  -->
    <!-- <node pkg="rosserial_python" type="serial_node.py" name="serial_node2">
        <param name="port" value="/dev/ttyACM0"/>
        <param name="baud" value="115200"/>
    </node> -->

    <node pkg="main" type="main" name="main">
        <param name="game_side" type="string" value="blue"/>
        <param name="camera_point_x" type="int" value="75"/>
        <param name="camera_point_y" type="int" value="200"/>
    </node>

    <!-- LINE DETECTION -->
    <node pkg="wall_detection" type="environment" name="environment"/>

    <!-- DETECTED BALL MSG READER -->
    <node pkg="camera" type="ball_detecter" name="ball_detecter" />

    <!-- KEY COMMAND WAITING NODE [s]=START, [r]=RETRY COMMAND -->
    <!-- <node pkg="main" type="kye_command.py" name="key_command.py"/> -->
</launch>